### PR TITLE
use a large enough buffer to flush in

### PR DESCRIPTION
**Pull request by Geal**
When writing a deferred response, if the output buffer was too small to write the entire compressed response, the compressor would write a small chunk, that did not decompress to the entire primary response, and would then wait for the next response to send the rest. Unfortunately, we cannot really know the output size we need in advance, and if we asked the decoder, it will tell us that it flushed all the data, even if it could have sent more.
So in here we raise the output buffer size, and do a second buffer growth step after flushing if necessary.

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

File: apollo-router/src/axum_factory/axum_http_server_factory.rs [hunk 1]
```diff
@@ -468,6 +468,7 @@ async fn handle_graphql(
         Ok(response) => {
             tracing::info!(counter.apollo_router_session_count_active = -1,);
             let (mut parts, body) = response.response.into_parts();
+            //println!("response parts before compression: {parts:?}");
```
Comment (bnjjj) [inline apollo-router/src/axum_factory/axum_http_server_factory.rs:471]: to delete

**Review by bnjjj — APPROVED** [review]
Just delete the println and we're good

File: apollo-router/src/axum_factory/compression/mod.rs [hunk 2]
```diff
@@ -94,24 +93,41 @@ where {
...
+                                            if !flushed {
+                                                if partial_output.unwritten().is_empty() {
+                                                    partial_output
+                                                        .extend(partial_output.written().len());
+                                                }
+                                                continue;
+                                            } else {
+                                                break;
+                                            }
+                                        }
```
Comment (o0Ignition0o) [inline apollo-router/src/axum_factory/compression/mod.rs:119]:
```suggestion
                                            if flushed {
                                                break;
                                            }
                                            if partial_output.unwritten().is_empty() {
                                                partial_output
                                                    .extend(partial_output.written().len());
                                            }
                                        }
```

**Review by o0Ignition0o — APPROVED** [review]

File: apollo-router/src/axum_factory/axum_http_server_factory.rs (hunk 1 above)
Comment (Geal) [inline apollo-router/src/axum_factory/axum_http_server_factory.rs:471]:
```suggestion
```

File: apollo-router/src/axum_factory/axum_http_server_factory.rs [hunk 3]
```diff
@@ -483,6 +484,7 @@ async fn handle_graphql(
                     Body::wrap_stream(compressor.process(body))
                 }
             };
+            //println!("response parts after compression: {parts:?}");
```
Comment (Geal) [inline apollo-router/src/axum_factory/axum_http_server_factory.rs:487]:
```suggestion
```

File: apollo-router/src/axum_factory/axum_http_server_factory.rs (hunk 1 above)
Comment (Geal) [reply apollo-router/src/axum_factory/axum_http_server_factory.rs]: fixed in [5bd4ab7](https://github.com/apollographql/router/pull/3067/commits/5bd4ab722d8169bd32ec6cd56948ef1cedb34207)

File: apollo-router/src/axum_factory/compression/mod.rs [hunk 4]
```diff
@@ -94,24 +93,41 @@ where {
                                 // there was not enough space in the output buffer to compress everything,
                                 // so we resize and add more data
                                 if partial_output.unwritten().is_empty() {
+                                    println!(
```
[the remark this answers is not in the document]
Comment (Geal) [reply apollo-router/src/axum_factory/compression/mod.rs]: fixed in [5bd4ab7](https://github.com/apollographql/router/pull/3067/commits/5bd4ab722d8169bd32ec6cd56948ef1cedb34207)

**Pull request merged**